core/src/main/scala/org/bitcoins/core/psbt/PSBT.scala

```diff
@@ -834,12 +829,12 @@ object PSBT extends Factory[PSBT] {
   }
 
   /**
-    * Wraps a Vector of pairs of NewSpendingInfos and the Transactions whose outputs are spent.
+    * Wraps a Vector of pairs of ScriptSignatureParams and the Transactions whose outputs are spent.
     * Note that this Transaction is only necessary when the output is non-segwit.
     */
   case class SpendingInfoAndNonWitnessTxs(
       infoAndTxOpts: Vector[
-        (ScriptSignatureParams[InputInfo], Option[BaseTransaction])]) {
+        (ScriptSignatureParams[InputInfo], Option[Transaction])]) {
```

Comment:
why the type change?

Reply:
because now witness txs need the previous tx

Reply:
Ah I see, in that case the name of the case class is going to have to change too :P

Reply:
Why is transaction an `Option[]` here? Also it might be better to define this type this way: 

```scala
case class SpendingInfoAndPrevTx(scriptSigParams: ScriptSignatureParams[InputInfo], txOpt: Option[Transaction])
```
and then you can just use `Vector[SpendingInfoAndPrevTx]` everywhere

Reply:
We use `Option` here because some won't need txs (future segwit versions). We could remove the `Option` until then but I don't see the point in that as we will just have to redo the work again.

Reply:
I guess my personal preference in these situations is to avoid using the `Option[]` type as it further complicates things, rather use an ADT to represent the different states we can have (getting rid of the `Option`!) and then in the future we can extend the type ADT to include segwit v1

Reply:
Was able to just remove this all together actually. Since our `ScriptSignatureParams` have a prexTx on them we don't need this. Cleaned a lot of things up :)